commandlineparser/CMakeLists.txt

```diff
@@ -1,2 +1,30 @@
-#CMAKE OF commandlineparser
-add_subdirectory(src)
\ No newline at end of file
+# setting relative paths for the install the directory
+set(include_dest include)
+set(lib_dest lib/${DBUILD_TYPE})
+
+
+#add the library
+add_library(cmd_lib src/Options.cpp src/Parser.cpp include/Parser.h include/Options.h)
```

Comment:
make a list of source and header file and pass it here

something like 

file(GLOB helloworld_SRC
    "*.h"
    "*.cpp"
)